### PR TITLE
[CI] Build a CPU-only wheel under name `xgboost-cpu`

### LINKED ISSUE #10596: Release a CPU only wheel on PyPI
I am using XGBoost on CPU only and do not require the NCCL library

Until before XGBoost v2.1.0 I managed to installed the lib with pip and --no-binay option to compile the package without NCCL

Since v2.1.0, `nvidia-nccl-cu12` is installed from PyPI and its huge 190MB weight makes my app not fit on my Heroku server anymore.

Is there another way to avoid the installation of this dependency than to edit the pyproject.toml file from XGBoost locally and install it locally ?

### PR DESCRIPTION
**Pull request by hcho3**
Closes #10596

Try to re-use code from https://github.com/dmlc/xgboost/pull/10478 and https://github.com/dmlc/xgboost/pull/10483

TODO
- [x] Build manylinux2014 wheels targeting x86_64 Linux and Windows
- [x] Build source distribution (sdist)
- [x] Update the doc and README

### CODE REVIEW

File: python-package/pyproject.toml [hunk 1]
```diff
@@ -7,13 +7,13 @@ build-backend = "packager.pep517"
...
-version = "2.2.0-dev"
+description = "XGBoost Python Package"
+readme = { file = "README.rst", content-type = "text/x-rst" }
 authors = [
     { name = "Hyunsu Cho", email = "chohyu01@cs.washington.edu" },
     { name = "Jiaming Yuan", email = "jm.yuan@outlook.com" }
 ]
-description = "XGBoost Python Package"
-readme = { file = "README.rst", content-type = "text/x-rst" }
+version = "2.2.0-dev"
```
Comment (hcho3) [inline python-package/pyproject.toml:16]: Moving this line here, so that the patch `cpu_only_pypkg.patch` doesn't contain the version number.

**Comment by hcho3** [comment]
@trivialfis This pull request is now ready for review.

**Review by trivialfis — APPROVED** [review]
Do we want to publish one for 2.1?

**Comment by asross** [comment]
Is it possible to build a wheel for mac as well?

**Comment by hcho3** [comment]
@asross That would not be necessary, since XGBoost already provides a CPU-only wheel targeting MacOS. More specifically, `
xgboost-2.1.2-py3-none-macosx_10_15_x86_64.macosx_11_0_x86_64.macosx_12_0_x86_64.whl` contains only CPU code.

**Pull request merged**